src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt

```diff
@@ -1003,16 +1003,40 @@ class IntelliJPlatformDependenciesHelper(
      * Creates a [Provider] that holds a JetBrains Runtime version obtained using the currently used IntelliJ Platform.
      */
     internal fun obtainJetBrainsRuntimeVersion() = cachedProvider {
-        val version = platformPath.get().resolve("dependencies.txt").takeIf { it.exists() }?.let {
+        val releaseFilePath = platformPath.get().resolve("jbr/release").takeIf { it.exists() }
+            ?: throw GradleException("Could not obtain JetBrains Runtime version with the current IntelliJ Platform.")
+
+        val properties = releaseFilePath.let {
             FileReader(it.toFile()).use { reader ->
                 with(Properties()) {
                     load(reader)
-                    getProperty("runtimeBuild") ?: getProperty("jdkBuild")
+                    this
                 }
             }
-        } ?: throw GradleException("Could not obtain JetBrains Runtime version with the current IntelliJ Platform.")
+        }
 
-        buildJetBrainsRuntimeVersion(version)
+        // JAVA_RUNTIME_VERSION property is missing in old versions.
+        // E.g JAVA_VERSION="17.0.6"
+        val javaVersion  = properties.getProperty("JAVA_VERSION")?.replace("\"", "")
+            ?: throw GradleException("Could not obtain JetBrains Runtime version with the current IntelliJ Platform.")
+        // E.g. IMPLEMENTOR_VERSION="JBR-17.0.6+1-653.34-jcef"
+        val implementorVersion  = properties.getProperty("IMPLEMENTOR_VERSION")?.replace("\"", "")
+            ?: throw GradleException("Could not obtain JetBrains Runtime version with the current IntelliJ Platform.")
+        // E.g. +1-653.34
+        val subString = implementorVersion.substringAfter(javaVersion).removeSuffix("-jcef").removeSuffix("-JCEF")
+        // E.g. 653.34-jcef
+        val buildVersion  = subString.replace(Regex("^([+][0-9]+-)"), "")
+        // E.g. 17.0.6-b653.34-jcef
+        // It is actually could be also "17.0.6-b829.4"
+        // https://github.com/JetBrains/JetBrainsRuntime/releases?page=8
+        // The latest are without -
+        // https://github.com/JetBrains/JetBrainsRuntime/releases
+        val fixedVersion = "${javaVersion}b${buildVersion}"
+
+        val isJcef = implementorVersion.endsWith("-jcef") || implementorVersion.endsWith("-JCEF")
+        val variant = if (isJcef) "jcef" else null
+
+        buildJetBrainsRuntimeVersion(fixedVersion, variant)
```

Comment:
This won't work on old IDEs:
IU-192.7142.36
![image](https://github.com/user-attachments/assets/45f5bc01-12c9-477d-9941-b741f9591973)

Reply:
Latest:
![image](https://github.com/user-attachments/assets/554d30b6-3d32-4e0e-9ec4-a4705c99bce6)

Reply:
Obsolete, as agreed